diff --git a/packages/cli/src/build/link.rs b/packages/cli/src/build/link.rs
index 2a1ff75b3..9f2c58a96 100644
--- a/packages/cli/src/build/link.rs
+++ b/packages/cli/src/build/link.rs
@@ -193,7 +193,7 @@ impl BuildRequest {
             .workspace_rustc
             .link_args
             .iter()
-            .filter(|arg| arg.ends_with(".rcgu.o"))
+            .filter(|arg| arg.ends_with(".rcgu.o") || arg.ends_with(".rcgu.asm.o"))
             .sorted()
             .map(PathBuf::from)
             .collect();
